src/cpu/aarch64/acl_gemm_convolution.hpp

```diff
@@ -105,13 +107,23 @@ struct acl_gemm_convolution_fwd_t : public primitive_t {
                 "gemm:acl", acl_gemm_convolution_fwd_t, USE_GLOBAL_SCRATCHPAD);
 
         status_t init(engine_t *engine) {
-            bool ok = true && is_fwd()
+            using namespace data_type;
+            using smask_t = primitive_attr_t::skip_mask_t;
+
+            bool ok = is_fwd()
                     && set_default_alg_kind(alg_kind::convolution_direct)
-                    && expect_data_types(data_type::f32, data_type::f32,
-                            data_type::f32, data_type::f32, data_type::f32)
+                    && expect_data_types(
+                            src_type, wei_type, bia_type, dst_type, undef)
                     && !has_zero_dim_memory()
-                    && attr()->has_default_values(
-                            primitive_attr_t::skip_mask_t::post_ops)
+                    && IMPLICATION(with_bias(),
+                            IMPLICATION(
+                                    src_type == u8, bias_md_.data_type == s32))
+                    && IMPLICATION(src_type == f32, bias_md_.data_type == f32)
+                    && attr()->has_default_values(smask_t::oscale
+                                    | smask_t::zero_points_runtime
+                                    | smask_t::post_ops,
```

Comment:
Could you please comment on how the post operations work in ACL for int8 convolution? Do they have the same sequence of operations as in oneDNN:

```
dst_s8 <- cast_with_saturation<s8>(
        post_op(
            output_scale_f32 * 
                (f32)( src_s8 (*) wei_s8 )
        )
    )
```

Reply:
Compute Library follows the TFLite quantization scheme.
As far as I've come to understand it, our int8 convolution first converts int8 values to int32, then does the multiplications and additions in int32, then scales down, adds bias if applicable, and then it applies any of the activation functions that are supported (only RELU-types) by clamping the values and then finally it does a saturating cast to the uint8 range.

This is in the case of a fused convolution layer. Other post-ops may not apply.

I hope I didn't misunderstand the question.

Reply:
@diaena Do scaling down and applying of activation happen in f32?

Reply:
@igorsafo All operations happen in the integer (int32) domain. We don't convert to floats.

Reply:
I see that QuantizationInfo(float scale, int offset) takes scale as f32 value. Will it be converted to s32 before applying to result of operation?

Reply:
to piggyback on @igorsafo question: If all computations happen in s32 and the scaling factor is converted to s32, is it represented in fixed point (so "arbitrary" scaling factors are supported), or do you accept only scaling factors of some form (e.g. powers of 2)?

Reply:
Hi both, so yes, indeed, internally Compute Library is converting the f32 scale value to a pair of (int fixedpoint_multiplier, int fixedpoint_shift). This pair is then passed on to the low precision gemm kernel.
I hope this explains it.

Reply:
Thank you for the clarification!